apps/dashboard/test/models/motd_formatter_markdown_erb_test.rb

```diff
@@ -50,5 +50,21 @@ class MotdTest < ActiveSupport::TestCase
 
     assert_not_nil formatted_motd.content
   end
+
+  test "motd-formatter-md-erb should recognize CurrentUserSingleton methods" do
+    path = "#{Rails.root}/test/fixtures/files/motd_current_user"
+    motd_file = MotdFile.new(path)
+    formatted_motd = MotdFormatterMarkdownErb.new(motd_file)
+
+    groups = OodSupport::Process.groups.map(&:name)
+
+    group  = OodSupport::Group.new
+    user_in_group = groups.include? group
+
+    expected_file = OodAppkit.markdown.render(
+      "You're in #{groups.size} groups\nincluding #{group}.")
+
+    assert_equal expected_file, formatted_motd.content
```

Comment:
I do not understand this test. IT doesn't use `CurrentUser.groups` but rather `groups = OodSupport::Process.groups.map(&:name)` and `group  = OodSupport::Group.new`

Reply:
Same for the other MOTD ERB tests

Reply:
CurrentUser.groups is in the fixture file "motd_current_user". I'm testing the functionality, making sure that CurrentUser.groups is returning the same value as would `OodSupport::Process.groups.map(&:name)`.

I do notice that I missed using user_in_group. So, I'll push a new commit fixing that.

Reply:
It doesn't make sense to me to test a method by adding the implementation of that method in the test (in this case `OodSupport::Process.groups.map(&:name)` happens to be the implementation).

Also these tests could be much shorter. If you want to test the CurrentUserSingleton methods themselves to confirm they return the correct values have a test that tests this directly. For example:

```ruby
test "CurrentUserSingleton#groups" do
  assert_equal Process.groups.count, CurrentUserSingleton.new.groups.count
end
```

As for the `test "motd-formatter-md-erb should recognize CurrentUserSingleton methods" do` all you need to do is render it. If `motd-formatter-md-erb` doesn't recognize `CurrentUserSingleton methods` then it will raise an exception.

```ruby
test "motd-formatter-md-erb should recognize CurrentUserSingleton methods" do
  refute_equal "", MotdFormatterMarkdownErb.new(MotdFile.new(File.join(fixtures_path, 'files', 'motd_current_user'))).content.to_s
end
```

Reply:
Okay, sounds good.  I'll push a new commit fixing the motd tests. As for the other tests, are those okay?